gateway/session_manager.go

```diff
@@ -418,3 +442,33 @@ func GetAccessDefinitionByAPIIDOrSession(session *user.SessionState, api *APISpe
 
 	return accessDef, allowanceScope, nil
 }
+
+type endpointRateLimitInfo struct {
+	keySuffix string
+	rate      float64
+	per       float64
+}
+
+func getEndpointRateLimitInfo(method string, path string, endpoints []user.Endpoint) (*endpointRateLimitInfo, bool) {
```

Comment:
Still think this should be moved. We have valid reasons to not extend gw scope

Reply:
moved as a receiver method on `Endpoints` type